modules/cfe_testcase/src/msg_api_test.c

```diff
@@ -0,0 +1,231 @@
+    UtAssert_True(size == 12u, "CFE_MSG_SetSize => size = 12u");
+
+    /* test get-type */
+    UtAssert_SUCCESS(CFE_MSG_GetType(&cmd.Msg, &type));
+    /* type has to be ...Type_Cmd because it created within a command */
+    UtAssert_INT32_EQ(type, CFE_MSG_Type_Cmd);
+
+    /* test msg set-type */
+    UtAssert_SUCCESS(CFE_MSG_SetType(&cmd.Msg, CFE_MSG_Type_Tlm));
+    /* check if set-type works like expected */
+    UtAssert_INT32_EQ(CFE_MSG_GetType(&cmd.Msg, &type), CFE_MSG_Type_Tlm);
+
+    /* test get header-version */
+    UtAssert_SUCCESS(CFE_MSG_GetHeaderVersion(&cmd.Msg, &hdrVer));
+    UtAssert_INT32_EQ(CFE_MSG_GetHeaderVersion(&cmd.Msg, NULL), CFE_MSG_BAD_ARGUMENT);
+
+    /* test set header-version */
+    UtAssert_SUCCESS(CFE_MSG_SetHeaderVersion(&cmd.Msg, 0));
+    CFE_MSG_GetHeaderVersion(&cmd.Msg, &hdrVer);
+    UtAssert_True(hdrVer == 0, "hdrVer = 0");
+    UtAssert_INT32_EQ(CFE_MSG_SetHeaderVersion(NULL, hdrVer), CFE_MSG_BAD_ARGUMENT);
+
+    /* test get-has-secondary-header */
+    UtAssert_SUCCESS(CFE_MSG_GetHasSecondaryHeader(&cmd.Msg, &hasSecondaryHdr));
+    UtAssert_INT32_EQ(CFE_MSG_GetHasSecondaryHeader(&cmd.Msg, NULL), CFE_MSG_BAD_ARGUMENT);
+
+    /* test set-has-scondary-header */
+    hasSecondaryHdr = true;
+    UtAssert_SUCCESS(CFE_MSG_SetHasSecondaryHeader(&cmd.Msg, hasSecondaryHdr));
+    hasSecondaryHdr = false;
+    CFE_MSG_GetHasSecondaryHeader(&cmd.Msg, &hasSecondaryHdr);
+    UtAssert_True(hasSecondaryHdr, "CFE_MSG_SetHasSecondaryHeader => CFE_MSG_GetHasSecondaryHeader = true");
+
+    /* test get-apid */
+    UtAssert_SUCCESS(CFE_MSG_GetApId(&cmd.Msg, &appId));
+    UtAssert_INT32_EQ(CFE_MSG_GetApId(&cmd.Msg, NULL), CFE_MSG_BAD_ARGUMENT);
+
+    /* test set-apid */
+    UtAssert_SUCCESS(CFE_MSG_SetApId(&cmd.Msg, 0));
+    UtAssert_True(CFE_MSG_GetApId(&cmd.Msg, &appId) == 0, "CFE_MSG_SetApId => apid = 0");
+    UtAssert_INT32_EQ(CFE_MSG_SetApId(&cmd.Msg, 0xFFFF), CFE_MSG_BAD_ARGUMENT);
+
+    /* test get/set-segmentation-flag */
+    UtAssert_INT32_EQ(CFE_MSG_GetSegmentationFlag(&cmd.Msg, NULL), CFE_MSG_BAD_ARGUMENT);
+
+    UtAssert_SUCCESS(CFE_MSG_SetSegmentationFlag(&cmd.Msg, CFE_MSG_SegFlag_Continue));
+    UtAssert_SUCCESS(CFE_MSG_GetSegmentationFlag(&cmd.Msg, &segFlag));
+    UtAssert_True(segFlag == CFE_MSG_SegFlag_Continue,
+                  "CFE_MSG_SetSegmentationFlag => segFlag = CFE_MSG_SegFlag_Continue");
+
+    /* test set/get-sequence-count */
+    UtAssert_INT32_EQ(CFE_MSG_GetSequenceCount(&cmd.Msg, NULL), CFE_MSG_BAD_ARGUMENT);
+
+    UtAssert_SUCCESS(CFE_MSG_SetSequenceCount(&cmd.Msg, 2u));
+    UtAssert_SUCCESS(CFE_MSG_GetSequenceCount(&cmd.Msg, &seqCnt));
+
+    UtAssert_True(seqCnt == 2u, "CFE_MSG_SetSequenceCount => seqCnt = 2");
+
+    /* test get-next-sequence-count */
+    /* 0x3FFF <- bad because defined CFE_MSG_SEQCNT_MASK value but not public */
+    UtAssert_True(CFE_MSG_GetNextSequenceCount(0x3FFF) == 0, "CFE_MSG_GetNextSequenceCount(0x3FFF) = 0");
```

Comment:
Yeah, probably better to just avoid checking implementation specific behavior within this test (it's private since API users shouldn't depend on it) since it is checked in the coverage test.  We could write a more implementation-specific functional check, but the intent of the current set of tests is to be implementation agnostic (in other words, test against the API documented behavior vs actual logic) where possible.